homeassistant/components/remootio/utils.py

```diff
@@ -0,0 +1,153 @@
+    result: str = ""
+
+    remootio_client: RemootioClient = None
+
+    try:
+        async with asyncio.timeout(REMOOTIO_TIMEOUT):
+            if not connection_options.connect_automatically:
+                remootio_client = await RemootioClient(
+                    connection_options,
+                    aiohttp_client.async_get_clientsession(hass),
+                    LoggerConfiguration(logger=logger),
+                )
+
+                await remootio_client.connect()
+            else:
+                remootio_client = await RemootioClient(
+                    connection_options,
+                    aiohttp_client.async_get_clientsession(hass),
+                    LoggerConfiguration(logger=logger),
+                )
+
+            if await _wait_for_connected(remootio_client):
+                await _check_sensor_installed(remootio_client)
+                await _check_api_version(remootio_client)
+
+                result = remootio_client.serial_number
+    finally:
+        if remootio_client is not None:
+            await remootio_client.terminate()
+
+    return result
+
+
+async def create_client(
+    hass: core.HomeAssistant,
+    connection_options: ConnectionOptions,
+    logger: Logger,
+    expected_serial_number: str | None = None,
+) -> RemootioClient:
+    """Create an Remootio client based on the given data."""
+    result: RemootioClient = None
+
+    try:
+        async with asyncio.timeout(REMOOTIO_TIMEOUT):
+            if not connection_options.connect_automatically:
+                try:
+                    result = await RemootioClient(
+                        connection_options,
+                        aiohttp_client.async_get_clientsession(hass),
+                        LoggerConfiguration(logger=logger),
+                    )
+                except BaseException as ex:
+                    raise ConfigEntryError(
+                        f'Failed to create client for Remootio device "{connection_options.host}"'
+                    ) from ex
+
+            try:
+                if connection_options.connect_automatically:
+                    result = await RemootioClient(
+                        connection_options,
+                        aiohttp_client.async_get_clientsession(hass),
+                        LoggerConfiguration(logger=logger),
+                    )
+                else:
+                    await result.connect()
+
+                if await _wait_for_connected(result):
+                    await _check_sensor_installed(result, False)
+                    await _check_api_version(result)
+
+                    if expected_serial_number is not None:
+                        serial_number: str = result.serial_number
+                        assert (
+                            expected_serial_number == serial_number
+                        ), f"Serial number of the Remootio device isn't the expected. Actual [{serial_number}] Expected [{expected_serial_number}]"
```

Comment:
Can this error happen in reality, for example if the IP-address has changed, or would we fail authentication in that case?